.github/workflows/docbuild.yaml

```diff
@@ -29,17 +29,19 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: 3.12
           cache-dependency-path: matter/docs/requirements.txt
           cache: pip
       - name: Install base dependencies
         working-directory: matter
         run: |
-          sudo pip3 install -U pip
-          pip3 install -r docs/requirements.txt
+          python3 -m venv venv
+          venv/bin/pip3 install -U pip
```

Comment:
NIT: How about skipping pip update? The pip version shipped with python should be good enough to install everything. If the pip is too old for something, it's rather an indication that the Python version is outdated...

Reply:
I agree, let's skip this.